### DISCUSSION TITLE
Inconsistent methods names `getRoleClass`, `getPermissionClass`

### DISCUSSION THREAD

**Discussion by erikn69** [spatie/laravel-permission#2367] [Q&A] [answered]
On `PermissionRegistrar` we have strings `$permissionClass, $roleClass`, and they are filled with strings from config
https://github.com/spatie/laravel-permission/blob/a7692219b322657f8ed97c376dc7f91ad7dd464f/src/PermissionRegistrar.php#L22-L26
https://github.com/spatie/laravel-permission/blob/a7692219b322657f8ed97c376dc7f91ad7dd464f/src/PermissionRegistrar.php#L66-L67
But on `getPermissionClass(), getRoleClass()` we get an instance of the contract instead of a string from getter
https://github.com/spatie/laravel-permission/blob/a7692219b322657f8ed97c376dc7f91ad7dd464f/src/PermissionRegistrar.php#L220-L226
So, for the method names to be more consistent, it should be something like this
```php
public function getPermissionClass(): string
{
    return $this->permissionClass;
}
public function getPermissionInstance(): Permission 
{
    return app(Permission::class);
}

public function getRoleClass(): string
{
    return $this->roleClass;
}
public function getRoleInstance(): Role 
{
    return app(Role::class);
}

```
----

Also on `clearClassPermissions()`, it doesn't clear `$permissionClass`, it clear `$permissions` propertie
https://github.com/spatie/laravel-permission/blob/a7692219b322657f8ed97c376dc7f91ad7dd464f/src/PermissionRegistrar.php#L153-L161

Ping @drbyte, @freekmurze

**Comment by erikn69** [comment]
Even worse on, `HasPermissions` trait we have a string `$permissionClass` propertie, but on `getPermissionClass()` package set an Permission contract instance on that propertie, same for `hasRolesTrait`
https://github.com/spatie/laravel-permission/blob/a7692219b322657f8ed97c376dc7f91ad7dd464f/src/Traits/HasPermissions.php#L20-L23
https://github.com/spatie/laravel-permission/blob/a7692219b322657f8ed97c376dc7f91ad7dd464f/src/Traits/HasPermissions.php#L42-L49
https://github.com/spatie/laravel-permission/blob/a7692219b322657f8ed97c376dc7f91ad7dd464f/src/PermissionRegistrar.php#L221-L223

**Comment by drbyte** [comment, marked as answer]
As discussed in #2368, thanks for your work on optimizing this. In addition to being more intuitive to both read/understand and use, it will improve performance as well.
